core/imageroot/var/lib/nethserver/cluster/actions/change-user-password/50update

```diff
@@ -0,0 +1,36 @@
+#!/usr/bin/env python3
+
+#
+# Copyright (C) 2021 Nethesis S.r.l.
+# http://www.nethesis.it - nethserver@nethesis.it
+#
+# This script is part of NethServer.
+#
+# NethServer is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License,
+# or any later version.
+#
+# NethServer is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with NethServer.  If not, see COPYING.
+#
+
+import sys
+import json
+import agent
+import redis
+
+request = json.load(sys.stdin)
+
+# Try to authenticate with user and password using unprivileged connection
+auth_db = redis.Redis(host='localhost', db=0, port=6379)
+if auth_db.execute_command('AUTH', request['user'], request['current_password']):
+    # On auhtentication success, switch to privileged connection to chang the password
+    rdb = agent.redis_connect(privileged=True)
+    rdb.acl_setuser(request['user'], passwords=[f'+{request["new_password"]}'], reset_passwords=True, enabled=True)
+    json.dump({}, fp=sys.stdout)
```

Comment:
I'd prefer returning a specific error code (and message). If the AUTH command fails I'd expect the UI behaves like a validation error occurs.

```suggestion
    json.dump({}, fp=sys.stdout)
else:
   # XXX validation error: invalid password
   sys.exit(2)
```

Reply:
Let's ask @andre8244

Reply:
Ok, but a step abort with a sys.exit(...) line is needed in any case. The current code seems to fail only due to a output validation error, isn't it?

Reply:
@andre8244 prefers a validation error, let's go with it.

> The current code seems to fail only due to a output validation error, isn't it?

The code will raise an exception and exit with 1 if `acl_setuser` fails

Reply:
Ok, that's a runtime error case. The point is the AUTH failure: if it fails that's a user-input error, thus I think it has to be handled like a validation error.